selftests/functional/test_runner_queue.py

```diff
@@ -0,0 +1,75 @@
+import os
+import sys
+import shutil
+import tempfile
+
+if sys.version_info[:2] == (2, 6):
+    import unittest2 as unittest
+else:
+    import unittest
+
+from avocado.core import exit_codes
+from avocado.utils import script
+from avocado.utils import process
+
+basedir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+basedir = os.path.abspath(basedir)
+
+AVOCADO_EXTERNAL_LIB = """
+def cleanup_function(arg1):
+    print('Bogus cleanup function. arg1: %s' % arg1)
+"""
+
+AVOCADO_TEST_RUNNER_QUEUE_EXTERNAL_LIB = """#!/usr/bin/env python
+from avocado import Test
+from avocado.utils import runtime as avocado_runtime
+from avocado_queue_testlib import cleanup_function
+
+
+class AvocadoQueueTests(Test):
+    def test_stuff_add_cleanup(self):
+        avocado_runtime.CURRENT_TEST.runner_queue.put({'func_at_exit': cleanup_function,
```

Comment:
How about `self.runner_queue`? I hope we remove `CURRENT_TEST` from the runtime as I really hate it.